### ISSUE TITLE
3.1.5: Cannot install a product when a newer version is installed.

### ISSUE THREAD

**Issue by Deilan** [dotnet/sdk#12853]
Can't install 3.1.5 (both 3.1.105 and 3.1.301) due to the following error:
Error 0x80070666: Cannot install a product when a newer version is installed.

![image](https://user-images.githubusercontent.com/4777512/89813268-17b4a900-db4a-11ea-9905-a35395ff2859.png)

But actually neither of them are installed:

![image](https://user-images.githubusercontent.com/4777512/89813315-2bf8a600-db4a-11ea-8ce6-87d1300a67b0.png)

<details>
  <summary>Installer log</summary>
  
  ```
  [2EE4:C078][2020-08-10T20:34:58]i001: Burn v3.10.4.4718, Windows v10.0 (Build 18362: Service Pack 0), path: C:\WINDOWS\Temp\{A2797EA9-AE7C-400E-8F63-DF7C07091445}\.cr\dotnet-sdk-3.1.105-win-x64.exe
[2EE4:C078][2020-08-10T20:34:58]i000: Initializing string variable 'DOTNETHOME_X86' to value '[ProgramFilesFolder]dotnet'
[2EE4:C078][2020-08-10T20:34:58]i000: Initializing string variable 'DOTNETHOME_X64' to value '[ProgramFiles64Folder]dotnet'
[2EE4:C078][2020-08-10T20:34:58]i000: Initializing string variable 'DOTNETHOME' to value '[DOTNETHOME_X64]'
[2EE4:C078][2020-08-10T20:34:58]i000: Initializing string variable 'BUNDLEMONIKER' to value 'Microsoft .NET Core SDK 3.1.105'
[2EE4:C078][2020-08-10T20:34:58]i000: Initializing string variable 'DOTNETSDKVERSION' to value '3.1.105'
[2EE4:C078][2020-08-10T20:34:58]i000: Initializing string variable 'DOTNETRUNTIMEVERSION' to value '3.1.5'
[2EE4:C078][2020-08-10T20:34:58]i000: Initializing string variable 'ASPNETCOREVERSION' to value '3.1.5'
[2EE4:C078][2020-08-10T20:34:58]i000: Initializing string variable 'WINFORMSANDWPFVERSION' to value '3.1.5'
[2EE4:C078][2020-08-10T20:34:58]i000: Initializing string variable 'DOTNETHOMESIMILARITYCHECKOVERRIDE' to value ''
[2EE4:C078][2020-08-10T20:34:58]i009: Command Line: '-burn.clean.room=D:\Downloads\Chrome\dotnet-sdk-3.1.105-win-x64.exe -burn.filehandle.attached=696 -burn.filehandle.self=700'
[2EE4:C078][2020-08-10T20:34:58]i000: Setting string variable 'WixBundleOriginalSource' to value 'D:\Downloads\Chrome\dotnet-sdk-3.1.105-win-x64.exe'
[2EE4:C078][2020-08-10T20:34:58]i000: Setting string variable 'WixBundleOriginalSourceFolder' to value 'D:\Downloads\Chrome\'
[2EE4:C078][2020-08-10T20:34:58]i000: Setting string variable 'WixBundleLog' to value 'C:\Users\Deilan\AppData\Local\Temp\Microsoft_.NET_Core_SDK_3.1.105_(x64)_20200810203458.log'
[2EE4:C078][2020-08-10T20:34:58]i000: Setting string variable 'WixBundleName' to value 'Microsoft .NET Core SDK 3.1.105 (x64)'
[2EE4:C078][2020-08-10T20:34:58]i000: Setting string variable 'WixBundleManufacturer' to value 'Microsoft Corporation'
[2EE4:8BE4][2020-08-10T20:34:58]i000: Setting numeric variable 'WixStdBALanguageId' to value 1033
[2EE4:8BE4][2020-08-10T20:34:58]i000: Setting version variable 'WixBundleFileVersion' to value '3.1.105.15039'
[2EE4:C078][2020-08-10T20:34:58]i100: Detect begin, 15 packages
[2EE4:C078][2020-08-10T20:34:58]i000: Setting numeric variable 'DotnetInstallLocationExists_x64' to value 1
[2EE4:C078][2020-08-10T20:34:58]i000: Setting numeric variable 'DotnetInstallLocationExists_x86' to value 1
[2EE4:C078][2020-08-10T20:34:58]i052: Condition 'DotnetInstallLocationExists_x64' evaluates to true.
[2EE4:C078][2020-08-10T20:34:58]i000: Setting string variable 'DOTNETHOME_X64' to value 'C:\Program Files\dotnet'
[2EE4:C078][2020-08-10T20:34:58]i052: Condition 'NOT DotnetInstallLocationExists_x64' evaluates to false.
[2EE4:C078][2020-08-10T20:34:58]i052: Condition 'DotnetExeExists_x64' evaluates to false.
[2EE4:C078][2020-08-10T20:34:58]i052: Condition 'DotnetInstallLocationExists_x86' evaluates to true.
[2EE4:C078][2020-08-10T20:34:58]i000: Setting string variable 'DOTNETHOME_X86' to value 'C:\Program Files (x86)\dotnet'
[2EE4:C078][2020-08-10T20:34:58]i052: Condition 'NOT DotnetInstallLocationExists_x86' evaluates to false.
[2EE4:C078][2020-08-10T20:34:58]i052: Condition 'DotnetExeExists_x86' evaluates to false.
[2EE4:C078][2020-08-10T20:34:58]i102: Detected related bundle: {5ab8e3af-c7f2-4178-9337-71d3c2d20ed8}, type: Upgrade, scope: PerMachine, version: 3.1.106.15050, operation: Downgrade
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: dotnet_runtime_3.1.5_win_x64.msi, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: dotnet_hostfxr_3.1.5_win_x64.msi, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: dotnet_host_3.1.5_win_x64.msi, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: windowsdesktop_runtime_3.1.5_win_x64.msi, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: dotnet_targeting_pack_3.1.0_win_x64.msi, state: Present, cached: Complete
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: dotnet_apphost_pack_3.1.5_win_x64.msi, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: dotnet_apphost_pack_3.1.5_win_x64_x86.msi, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: dotnet_apphost_pack_3.1.5_win_x64_arm.msi, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: dotnet_apphost_pack_3.1.5_win_x64_arm64.msi, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: netstandard_targeting_pack_2.1.0_win_x64.msi, state: Present, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: aspnetcore_targeting_pack_3.1.3_win_x64.msi, state: Present, cached: Complete
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: windowsdesktop_targeting_pack_3.1.0_win_x64.msi, state: Present, cached: Complete
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: dotnet_31templates_3.1.105_servicing_015039_win_x64.msi, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: dotnet_sdk_internal_3.1.105_win_x64.msi, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i101: Detected package: AspNetCoreSharedFramework_x64, state: Absent, cached: None
[2EE4:C078][2020-08-10T20:34:58]i052: Condition 'WixBundleInstalled OR ((NOT (DOTNETHOME_X64 ~= DOTNETHOME_X86)) OR DOTNETHOMESIMILARITYCHECKOVERRIDE)' evaluates to true.
[2EE4:C078][2020-08-10T20:34:58]i199: Detect complete, result: 0x0
[2EE4:8BE4][2020-08-10T20:35:00]e000: Error 0x80070666: Cannot install a product when a newer version is installed.

  ```
  
</details>

**Comment by wli3** [comment]
@joeloff could you look into this?

**Comment by joeloff** [comment]
Yup, I'll take a look

**Comment by joeloff** [comment]
@Deilan Starting in 3.0, SDKs installations can be upgraded within their feature band. So 3.1.100 can be upgraded to any higher 3.1.1xx value (3.1.101, 3.1.102, etc.) However, if you have an SDK installed, you cannot install an SDK of the same band, but a lower patch level.

In your case, it looks like you have 3.1.106, so you can't install 3.1.105 because that's a downgrade. The same is true for the 3.1.301 install (you already have 3.1.302).

**Comment by Thaina** [comment]
Also occur in `3.1.402` which is the latest possible I could download from https://dotnet.microsoft.com/download/

**Comment by glennamanns** [comment]
I found myself in need of 3.1.401 (x64) and I had 3.1.402 (x64) installed. I accidentally installed 3.1.401 x86 (this completed successfully). After uninstalling that, I attempted to install 3.1.401 x64, and kept running into this error. I assumed something had been left behind by the 3.1.401 x86 installation and spent a long time looking for leftover files until I found this GitHub issue.

These are definitely rookie errors of mine, but it would have been really helpful if the error message indicated that downgrading within band is not supported.
